src/components/Settings/MaxSlippageSettings/index.tsx

```diff
@@ -0,0 +1,174 @@
+import { Trans } from '@lingui/macro'
+import { Percent } from '@uniswap/sdk-core'
+import Expand from 'components/Expand'
+import QuestionHelper from 'components/QuestionHelper'
+import Row, { RowBetween } from 'components/Row'
+import React, { useState } from 'react'
+import { useUserSlippageTolerance } from 'state/user/hooks'
+import { SlippageTolerance } from 'state/user/types'
+import styled from 'styled-components/macro'
+import { CautionTriangle, ThemedText } from 'theme'
+
+import { Input, InputContainer } from '../Input'
+
+enum SlippageError {
+  InvalidInput = 'InvalidInput',
+}
+
+const Option = styled(Row)<{ isActive: boolean }>`
+  width: auto;
+  cursor: pointer;
+  padding: 6px 12px;
+  text-align: center;
+  gap: 4px;
+  border-radius: 12px;
+  background: ${({ isActive, theme }) => (isActive ? theme.backgroundInteractive : 'transparent')};
+  pointer-events: ${({ isActive }) => isActive && 'none'};
+`
+
+const Switch = styled(Row)`
+  width: auto;
+  padding: 4px;
+  border: 1px solid ${({ theme }) => theme.backgroundOutline};
+  border-radius: 16px;
+`
+
+const NUMBER_WITH_MAX_TWO_DECIMAL_PLACES = /^(?:\d*\.\d{0,2}|\d+)$/
+const MINIMUM_RECOMMENDED_SLIPPAGE = new Percent(5, 10_000)
+const MAXIMUM_RECOMMENDED_SLIPPAGE = new Percent(1, 100)
+
+export default function MaxSlippageSettings({ autoSlippage }: { autoSlippage: Percent }) {
+  const [userSlippageTolerance, setUserSlippageTolerance] = useUserSlippageTolerance()
+
+  // In order to trigger `custom` mode, we need to set `userSlippageTolerance` to a value that is not `auto`.
+  // To do so, we use `autoSlippage` value. However, since users are likely to change that value,
+  // we render it as a placeholder instead of a value.
+  const defaultSlippageInputValue =
```

Comment:
Call this `placeholderSlippageValue`

Reply:
I find `placeholder` name confusing in this context, because this value is not being used as a `placeholder` for the Input. This value is used as a `Input` value, showing the slippage value, except for default and auto mode.